addons/settings/test_parse.sqf

```diff
@@ -0,0 +1,94 @@
+// ----------------------------------------------------------------------------
+#define DEBUG_MODE_FULL
+#include "script_component.hpp"
+
+SCRIPT(test_settings);
+
+// ----------------------------------------------------------------------------
+
+private ["_funcName", "_settings", "_result"];
+
+LOG('Testing Settings');
+
+// UNIT TESTS (parse)
+_funcName = QFUNC(parse);
+TEST_DEFINED(QFUNC(parse),"");
+
+// Purposely weird formatting, must remain this way because newlines remain part of result
+_settings = (preprocessFile "x\cba\addons\settings\test_settings_regular.sqf") call FUNC(parse);
+_result = _settings isEqualTo [
+    ["ace_advanced_ballistics_ammoTemperatureEnabled", true, 0],
+    ["ace_advanced_ballistics_barrelLengthInfluenceEnabled", false, 2],
+    ["ace_advanced_ballistics_bulletTraceEnabled", true, 1],
+    ["ace_advanced_fatigue_enabled", false, 0],
+    ["ace_advanced_fatigue_enableStaminaBar", true, 0],
+    ["ace_advanced_fatigue_performanceFactor", 1, 1],
+    ["ace_advanced_fatigue_terrainGradientFactor", 1, 2]
+];
+TEST_TRUE(_result,_funcName);
+
+_settings = (preprocessFile "x\cba\addons\settings\test_settings_multiline.sqf") call FUNC(parse);
+_result = _settings isEqualTo [
+    ["test1", "[
+    ""  item_1  "",
+    ""  item_2  ""
+]", 0],
+    ["test2", "[
+    '  item_1  ',
+            '  item_2  '
+]", 1],
+    ["test3", "[
+    '  item_1  '   , ""  item_2  ""
+]", 2],
+    ["test4", "
+
+[
+'""  item_1  ""',
+'  item_2  '
+]
+
+", 0]
+];
+TEST_TRUE(_result,_funcName);
+
+_settings = (preprocessFile "x\cba\addons\settings\test_settings_unicode.sqf") call FUNC(parse);
+_result = _settings isEqualTo [["test1", "[Āā, Ăă, Ҙ, привет]", 1]];
+TEST_TRUE(_result,_funcName);
+
+_settings = (preprocessFile "x\cba\addons\settings\test_settings_strings.sqf") call FUNC(parse);
+_result = _settings isEqualTo [
+	["test1", "", 0],
+	["test2", "", 0],
+	["test3", "  T E S T  " , 0],
+	["test4", "  T E S T  ", 0],
+	["test5", "[ '  t e s t  ' , ""  T E S T  "" ]", 0],
+	["test6", "[ ""  t e s t  "" ,
+
+""""  T E S T  """" ]", 0],
+	["test7", "[ true, false ]", 0],
+	["test8", "[ ""  item_1  "" , ""  item_2  "" ]", 0],
+	["test9", "[ '  item_1  ' , '  item_2  ' ]", 0],
+	["test10", "[ '  item_1  ' , ""  item_2  "" ]", 0],
+	["test11", "[ '""  item_1  ""' , '  item_2  ' ]", 0],
+	["test12", "[ '  item_1  ' , ""'  item_2  '"" ]", 0]
```

Comment:
Tabs!

Reply:
> Tabs!

Tabs inside or outside of the strings? I saw the github action complained about tabs, but I want the tests to run with tabs (to make sure they work - unless that's unnecessary?).

Reply:
Oh right, I see now. @PabstMirror do we have a way to ignore validating parts of code?

I don't think we _need_ to test for tabs though, as whitespace is whitespace to Arma, but I may be wrong.

Reply:
passing CI now